src/tpm/tpm_wrapper.rs

```diff
@@ -86,78 +60,77 @@ pub fn public_key(key_path: &str) -> Result<Vec<u8>> {
         let mut blob_sz: tss2::size_t = 0;
         let mut offset: tss2::size_t = 0;
         let c_path = CString::new(key_path.as_bytes()).unwrap();
+
         tss2_call!(Fapi_GetEsysBlob(
             *tpm_ctx,
             c_path.as_ptr(),
             &mut blob_type as *mut u8,
             &mut esys_blob as *mut *mut u8,
             &mut blob_sz as *mut tss2::size_t,
         ))?;
+        let esys_blob = guard(esys_blob, |p| libc::free(p as *mut c_void));
+
         if blob_type != FAPI_ESYSBLOB_CONTEXTLOAD as u8 {
             return Err(TpmError::wrong_key_path());
         }
 
         let mut key_context: MaybeUninit<TPMS_CONTEXT> = MaybeUninit::uninit();
         tss2_call!(Tss2_MU_TPMS_CONTEXT_Unmarshal(
-            esys_blob,
+            *esys_blob,
             blob_sz,
             &mut offset as *mut tss2::size_t,
             key_context.as_mut_ptr(),
-        ), always => {
-            libc::free(esys_blob as *mut c_void)
-        })?;
+        ))?;
 
+        // TODO: what exactly does this mean?
+        //
+        // > Once the application finishes its direct interaction with
+        // > the TPM it SHALL delete its pointer to the TCTI context
+        //
+        // Are we responsible for freeing `tcti_ctx`?
```

Comment:
nope. GetTcti doesn't allocate new tcti context. It returns pointer to inner structure of TPM context. no need to free it.